### DISCUSSION TITLE
Extension or Subclass ScatterPlatLayer to draw Ellipse

### DISCUSSION THREAD

**Discussion by TextTL** [visgl/deck.gl#8601] [Q&A]
I want to draw Ellipses and Circle using  GeoJsonLayer,  what method is appropriate? Subclass or Extension of ScatterPlotLayer ? And any suggestion to modify ScatterPlotLayer to draw both Ellipse, Circle and Arc?

Thanks,

**Comment by grahambates-code** [comment]
i use turf.js for this in a Composite Layer that generates the geojson once  and sets it in state

**Comment by Pessimistress** [comment]
- GeoJsonLayer with `pointType: 'icon'` will let you draw a limited set of shapes
- If you want to control the aspect ratio of your eclipses you need to create a new attribute to pass additional data to the shader. Check out https://observablehq.com/@pessimistress/deck-gl-custom-layer-by-subclassing

**Reply by TextTL** [reply]
Thank you, this very helpful.
